mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -2466,7 +2467,7 @@ pub async fn taker_swap_trade_preimage(
         .mm_err(|e| TradePreimageRpcError::from_trade_preimage_error(e, other_coin_ticker))?;
 
     let prepared_params = TakerSwapPreparedParams {
-        dex_fee: dex_amount,
+        dex_fee: dex_amount.fee_amount(),
```

Comment:
`check_balance_for_taker_swap` should check the balance for the `total_spend_amount`